### ISSUE TITLE
Proper syntax for creating Sonos presets

### ISSUE THREAD

**Issue by u8915055** [kalkih/mini-media-player#317]
Hello, Ive been trying everything I can think of to be able to create a preset for a sonos device.
I use spotify.

Im trying to do any of the following:

1. preset pointing to a sonos favorite
2. preset pointing to a sonos playlist
3. preset pointing to a sonos music library album or song
4. preset pointing to a spotify playlist - not sure if I solve #1 this may not be an issue.  Thinking that if I set a spotify playlist as a sonos favorite maybe I can make it work, but have to solve #1 first.

For example, the syntax shown in the samples "spotify:user:spotify:playlist:37i9dQZF1DZ06evO2O09Hg"
This does not work for me.

Has anyone gotten this to work with sonos?

I have been able to get say an MP3 streaming station working through x-rincon-mp3radio.

I do see a previx x-sonos-spotify but i can only seem to get this to work with a single track, not a playlist or album.

Any help greatly appreciated. This player is quite amazing, just want to fully use it.

**Comment by u8915055** [comment]
I wanted to add a bit more information that I found with a sniffer.

First off, it seems that whether the item you want to play is a favorite or not doesnt matter.  A call to a spotify playlist whether as a sonos favorite or not doesnt seem to change the call.

So, this is what im finding with a sniffer.

Calling a spotify playlist yields the following call:

<EnqueuedURI>x-rincon-cpcontainer:1006206cspotify%3aplaylist%3a37i9dQZF1DZ06evO3SCFWM?sid=12&amp;flags=8300&amp;sn=11</EnqueuedURI>

If I create a shortcut config for this URI I get:
```
- name: Oscar
  type: music
  id: x-rincon-cpcontainer:1006206cspotify%3aplaylist%3a37i9dQZF1DZ06evO3SCFWM?sid=12&flags=8300&sn=11
```
But this does not seem to work.  It does nothing.

Next I called a streaming radio station that I had saved as a favorite in Sonos.  This is the call:
<CurrentURI>x-sonosapi-stream:s297990?sid=254&amp;flags=8224&amp;sn=0</CurrentURI>

I created a shortcut config for this:
```
- name: MSNBC
  type: music
  id: x-sonosapi-stream:s297990?sid=254&flags=8224&sn=0
```
This also did not work.

I then tried to play an album that I had in my sonos (local) library.  This was the call:
<EnqueuedURI>x-rincon-playlist:RINCON_5CAAFD546F7201400#A:ALBUMARTIST/Four80East/En%20Route</EnqueuedURI>

And the corresponding shortcut config for this:
```
- name: Four80East
   type: music
   id: x-rincon-playlist:RINCON_5CAAFD546F7201400#A:ALBUMARTIST/Four80East/En%20Route
```
This actually DOES work

I still have to test a sonos playlist.

So there seems to be some sort of issue with the spotify playlist and the streaming station.

Anyone have any ideas?

**Comment by u8915055** [comment]
Ok I found something . . . duh!

If you use the type of source, then the ID just needs to be the name of something that's in your favorites in the Sonos app.  For example.  In my favorites I have this spotify playlist "This is Oscar Peterson", so I used:

```
- name: Four80East2
   type: source
   id: This Is Oscar Peterson
```
And it works..
I also realize my issue is more to do with the HASS media player component and not this media player itself (I believe).

hope this is helpful to someone.  Maybe this can be included in the docs.
